src/web/event-notifications/event-notification-types/SlackNotificationForm.jsx

```diff
@@ -0,0 +1,167 @@
+
+class SlackNotificationForm extends React.Component {
+    static propTypes = {
+      config: PropTypes.object.isRequired,
+      validation: PropTypes.object.isRequired,
+      onChange: PropTypes.func.isRequired,
+    };
+
+    static defaultConfig = {
+      color: '#FF0000',
+      webhook_url: '',
+      channel: '#channel',
+      /* eslint-disable no-template-curly-in-string */
+      custom_message: ''
+                      + '--- [Event Definition] ---------------------------\n'
+                      + 'Title:       ${event_definition_title}\n'
+                      + 'Type:        ${event_definition_type}\n'
+                      + '--- [Event] --------------------------------------'
+                      + 'Timestamp:            ${event.timestamp}'
+                      + 'Message:              ${event.message}'
+                      + 'Source:               ${event.source}'
+                      + 'Key:                  ${event.key}'
+                      + 'Priority:             ${event.priority}'
+                      + ' Alert:                ${event.alert}'
+                      + 'Timestamp Processing: ${event.timestamp}'
+                      + 'Timerange Start:      ${event.timerange_start}'
+                      + 'Timerange End:        ${event.timerange_end}'
+                      + 'Event Fields:'
+                      + '${foreach event.fields field}'
+                      + '${field.key}: ${field.value}'
+                      + '${end}'
+                      + '${if backlog}'
+                      + '--- [Backlog] ------------------------------------'
+                      + 'Last messages accounting for this alert:'
+                      + '${foreach backlog message}'
+                      + '${message.id} -${message.timestamp} - ${message.source}'
+                      + '${message.message}'
+                      + '${end}${else}<No backlog>\n'
+                      + '${end}',
+      /* eslint-enable no-template-curly-in-string */
+      user_name: 'Graylog',
+      notify_channel: false,
+      link_names: false,
+      icon_url: '',
+      icon_emoji: '',
+
+    };
+
+    propagateChange = (key, value) => {
+      const { config, onChange } = this.props;
+      const nextConfig = lodash.cloneDeep(config);
+      nextConfig[key] = value;
+      onChange(nextConfig);
+    };
+
+    handleColorChange = (color, _, hidePopover) => {
+      hidePopover();
+      this.propagateChange('color', color);
+    };
+
+    handleChange = (event) => {
+      const { name } = event.target;
+      this.propagateChange(name, FormsUtils.getValueFromInput(event.target));
+    };
+
+    render() {
+      const { config, validation } = this.props;
+
+      return (
+        <>
+
+          <FormGroup controlId="color">
+            <ControlLabel>Configuration color</ControlLabel>
+            <div>
+              <ColorLabel color={config.color || '#f06292'} />
```

Comment:
If `config` is required, then we should not need defaults here

Reply:
ok let me try that.